compatibility_test_sim/tests/integration/compatibility.rs

```diff
@@ -0,0 +1,188 @@
+// Copyright © 2023 HQS Quantum Simulations GmbH. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the
+// License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing permissions and
+// limitations under the License.
+
+use roqoqo::operations::*;
+
+use test_roqoqo_1_2::backends::EvaluatingBackend;
+
+use qoqo_calculator::CalculatorFloat;
+
+use roqoqo_quest::Backend;
+
+use ndarray::{array, Array1};
+use num_complex::{Complex, Complex64};
+use test_case::test_case;
+
+// helper function
+fn is_close(a: Complex64, b: Complex64) -> bool {
+    (a - b).norm() < 1e-10
+}
+
+#[test_case((false, false, false), 0)]
+#[test_case((false, false, true), 1)]
+#[test_case((false, true, false), 2)]
+#[test_case((false, true, true), 3)]
+#[test_case((true, false, false), 4)]
+#[test_case((true, false, true), 5)]
+#[test_case((true, true, false), 6)]
+#[test_case((true, true, true), 7)]
+fn test_circuits_3q(iteration: (bool, bool, bool), value: usize) {
+    let ccz = ControlledControlledPauliZ::new(0, 1, 2);
+    let ccps = ControlledControlledPhaseShift::new(0, 1, 2, CalculatorFloat::from(0.3));
+    let unitary_ccz = ccz.unitary_matrix().unwrap();
+    let unitary_ccps = ccps.unitary_matrix().unwrap();
+
+    // Preparing matrices for later simulation
+    let c0: Complex64 = Complex::new(0.0, 0.0);
+    let c1: Complex64 = Complex::new(1.0, 0.0);
+    let mut final_matrix: Array1<Complex64> = array![c0, c0, c0, c0, c0, c0, c0, c0];
+    final_matrix[value] = c1;
+    let final_matrix_ccz = final_matrix.dot(&unitary_ccz);
+    let final_matrix_ccps = final_matrix.dot(&unitary_ccps);
```

Comment:
That multiplies the initial state_vector to the unitaries from the left, but we should multiply from the right.
Maybe it would be clearer to rename to initial_state_vector and final_state_vector
```
let final_state_vector_ccz = unitary_ccz.dot(initial_state_vector)
```